apps/condo/domains/organization/hooks/useTableColumns.tsx

```diff
@@ -1,110 +1,126 @@
+import { IFilters } from '@condo/domains/organization/utils/helpers'
+import { OrganizationEmployeeRole } from '@condo/domains/organization/utils/clientSchema'
+import { getOptionFilterDropdown } from '@condo/domains/common/components/Table/Filters'
+import { getFilterDropdownByKey } from '@condo/domains/common/utils/filters.utils'
+import { parseQuery } from '@condo/domains/common/utils/tables.utils'
+import { OrganizationEmployeeSpecialization } from '@condo/domains/organization/utils/clientSchema'
+import { getEmployeeSpecializationsMessage } from '@condo/domains/organization/utils/clientSchema/Renders'
 
 export const useTableColumns = (
+    filterMetas,
     organizationId: string,
-    sort: Array<string>,
-    filters: IFilters,
-    setFiltersApplied: React.Dispatch<React.SetStateAction<boolean>>
+    employees
 ) => {
     const intl = useIntl()
     const NameMessage = intl.formatMessage({ id: 'pages.auth.register.field.Name' })
     const RoleMessage = intl.formatMessage({ id: 'employee.Role' })
     const PositionMessage = intl.formatMessage({ id: 'employee.Position' })
     const PhoneMessage =  intl.formatMessage({ id: 'Phone' })
-    const EmailMessage = intl.formatMessage({ id: 'field.EMail' })
+    const SpecializationsMessage = intl.formatMessage({ id: 'employee.Specializations' })
 
-    const sorterMap = createSorterMap(sort)
-    const { loading, objs: organizationEmployeeRoles } = OrganizationEmployeeRole.useObjects({ where: { organization: { id: organizationId } } })
-    const search = getFilteredValue<IFilters>(filters, 'search')
+    const router = useRouter()
+    const { filters } = parseQuery(router.query)
+    const search = getFilteredValue(filters, 'search')
 
     const render = getTableCellRenderer(search)
 
+    const { objs: organizationEmployeeSpecializations } = OrganizationEmployeeSpecialization.useObjects({
+        where: {
+            employee: { id_in: employees.map(employee => employee.id) },
+        },
+    })
+
+    const { loading, objs: organizationEmployeeRoles } = OrganizationEmployeeRole.useObjects({ where: { organization: { id: organizationId } } })
+
     const renderCheckboxFilterDropdown = ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => {
         const adaptedStatuses = organizationEmployeeRoles.map(OrganizationEmployeeRole.convertGQLItemToFormSelectState).filter(identity)
         const filterProps = {
             setSelectedKeys,
             selectedKeys,
             confirm,
             clearFilters,
-            beforeChange: () => { setFiltersApplied(true) },
         }
 
         return getOptionFilterDropdown(adaptedStatuses, loading)(filterProps)
     }
 
+    const renderSpecializations = (employee) => {
+        const specializationsMessage = getEmployeeSpecializationsMessage(intl, employee, organizationEmployeeSpecializations)
+
+        return (
+            <Typography.Paragraph key={employee.id} style={{ margin: 0 }}>
```

Comment:
it's better to exclude inline styles

This project grew up like a ****, i think it's time to think about cases like this